control/control.go

```diff
@@ -848,15 +829,26 @@ func (p *pluginControl) FetchMetrics(ns core.Namespace, version int) ([]core.Cat
 }
 
 func (p *pluginControl) GetMetric(ns core.Namespace, ver int) (core.CatalogedMetric, error) {
-	return p.metricCatalog.Get(ns, ver)
+	return p.metricCatalog.GetMetric(ns, ver)
+}
+
+func (p *pluginControl) GetMetrics(ns core.Namespace, ver int) ([]core.CatalogedMetric, error) {
+	mts, err := p.metricCatalog.GetMetrics(ns, ver)
+	if err != nil {
+		return nil, err
+	}
+	rmts := make([]core.CatalogedMetric, len(mts))
```

Comment:
It has to stay here. I cannot move that to p.metriccatalog.GetMetrics() becuse that will remove information about plugin which is keep in metricType and available to retrieve form it.
Why:
- metricType includes core.CatalogedMetricType
- core.CatalogedMetricType does NOT include metricType

This "for" loop is only needed for a slice. For a single metricType there is no problem to return that directly.

See how it looks on snap master [snap/blob/master/control/control.go#L860](https://github.com/intelsdi-x/snap/blob/master/control/control.go#L860).